CONTRIBUTING.md

```diff
@@ -0,0 +1,393 @@
+- [Code Style Guidelines](#code-style-guidelines)
+- [Testing Guidelines](#testing-guidelines)
+- [Pull Request Process](#pull-request-process)
+- [Reporting Bugs](#reporting-bugs)
+- [Proposing Features](#proposing-features)
+- [Commit Message Conventions](#commit-message-conventions)
+- [Questions?](#questions)
+
+## Code of Conduct
+
+This project adheres to a code of conduct that all contributors are expected to follow. Please be respectful and constructive in all interactions.
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+If you find a bug, please report it by [opening an issue](https://github.com/florisvb/PyNumDiff/issues/new). See the [Reporting Bugs](#reporting-bugs) section for details.
+
+### Suggesting Enhancements
+
+Have an idea for a new feature or improvement? Please [open an issue](https://github.com/florisvb/PyNumDiff/issues/new) to discuss it. See the [Proposing Features](#proposing-features) section for details.
+
+### Contributing Code
+
+1. Look for issues labeled `good first issue` if you're new to the project
+2. Fork the repository
+3. Create a branch for your changes
+4. Make your changes following our guidelines
+5. Submit a pull request
+
+## Development Setup
+
+### Prerequisites
+
+- Python 3.7 or higher
+- Git
+- (Optional) A virtual environment manager (venv, conda, etc.)
+
+### Setting Up the Development Environment
+
+1. **Fork the repository** on GitHub
+
+2. **Clone your fork** locally:
+   ```bash
+   git clone https://github.com/YOUR_USERNAME/PyNumDiff.git
+   cd PyNumDiff
+   ```
+
+3. **Add the upstream repository**:
+   ```bash
+   git remote add upstream https://github.com/florisvb/PyNumDiff.git
+   ```
+
+4. **Create a virtual environment** (recommended):
+   ```bash
+   # Using venv
+   python -m venv venv
+   
+   # Activate on Windows
+   venv\Scripts\activate
+   
+   # Activate on macOS/Linux
+   source venv/bin/activate
+   ```
+
+5. **Install the package in development mode**:
+   ```bash
+   pip install -e .
+   ```
+
+6. **Install development dependencies**:
+   ```bash
+   pip install pytest pylint
+   ```
+
+7. **Verify the installation**:
+   ```bash
+   pytest -s
+   ```
+
+### Project Structure
+
+- `pynumdiff/` - Main source code
+- `examples/` - Jupyter notebook examples
+- `docs/` - Documentation source files
+- `.github/workflows/` - GitHub Actions CI configuration
+- `tests/` - Test files (if applicable)
+
+## Code Style Guidelines
+
+### Python Style Guide
+
+PyNumDiff follows [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guidelines. Here are the key points:
+
+- Use 4 spaces for indentation (no tabs)
+- Maximum line length: 79 characters (soft limit) or 99 characters (hard limit)
+- Use meaningful variable and function names
+- Follow naming conventions:
+  - Functions and variables: `snake_case`
+  - Classes: `PascalCase`
+  - Constants: `UPPER_SNAKE_CASE`
+
+### Code Quality
+
+The project uses `pylint` for code quality checks. Before submitting a PR:
```

Comment:
I haven't been doing this for a while either. But the plan is to submit this thing to JOSS again, and I think they're more strict about linting, so I'll have to run it again.